### ISSUE TITLE
Allow text in ignore comments

### ISSUE THREAD

**Issue by avazhnov-twilio** [hadolint/hadolint#1028]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

Currently such syntax is supporting by `hadolint` for [Inline ignores](https://github.com/hadolint/hadolint#inline-ignores):

```sh
# hadolint ignore=DL3003,SC1035
```

When I use Shellcheck, I prefer expanding a comment with text description, which is more human-readable:

```sh
# shellcheck disable=SC1091  # File not included in mock
```

Can we have similar logic for `hadolint`: to ignore whitespaces and custom text (for example, starting with `#`) in the same line after `ignore=ID1,ID2`?

### Expected behavior

Line

```Dockerfile
# hadolint ignore=SC1091  # File not included in mock
```

makes `hadolint` ignore `SC1091` in the next line.

### Actual behavior

Code

```Dockerfile
# hadolint ignore=SC1091  # File not included in mock
RUN python3 -m venv .venv && . .venv/bin/activate && pip3 install --no-cache-dir -r requirements.txt
```

gives a message like: "Dockerfile:7 SC1091 info: Not following: File not included in mock."

### Steps to reproduce the behavior

Create a `Dockerfile` like

```Dockerfile
FROM python:3

RUN groupadd -r myapp && useradd --no-log-init --system --gid myapp --create-home myapp
USER myapp

# hadolint ignore=SC1091  # File not included in mock
RUN python3 -m venv .venv && . .venv/bin/activate && pip3 install --no-cache-dir -r poetry
```

Check it with `hadolint`, it will give you a message: "Dockerfile:7 SC1091 info: Not following: File not included in mock."

**Output of `hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
# hadolint ignore=SC1091  # File not included in mock
RUN python3 -m venv .venv && . .venv/bin/activate && pip3 install --no-cache-dir -r requirements.txt
```

**Additional environment details (OS, stack version, etc.)**

Probably my request may also close https://github.com/hadolint/hadolint/issues/997.
